zkevm-circuits/benches/state_prover.rs

```diff
@@ -0,0 +1,163 @@
+    >;
+    type FloorPlanner = SimpleFloorPlanner;
+
+    fn without_witnesses(&self) -> Self {
+        Self::default()
+    }
+
+    fn configure(meta: &mut ConstraintSystem<F>) -> Self::Config {
+        Config::configure(meta)
+    }
+
+    fn synthesize(
+        &self,
+        config: Self::Config,
+        mut layouter: impl Layouter<F>,
+    ) -> Result<(), Error> {
+        config.load(&mut layouter)?;
+        config.assign(
+            layouter,
+            self.memory_ops.clone(),
+            self.stack_ops.clone(),
+            self.storage_ops.clone(),
+        )?;
+
+        Ok(())
+    }
+}
+
+fn bus_mapping_benchmark(c: &mut Criterion) {
+    let k = 14;
+    let input_trace = r#"
+        [
+            {
+                "pc": 5,
+                "op": "PUSH1",
+                "gas": 82,
+                "gasCost": 3,
+                "depth": 1,
+                "stack": [],
+                "memory": [
+                  "0000000000000000000000000000000000000000000000000000000000000000",
+                  "0000000000000000000000000000000000000000000000000000000000000000",
+                  "0000000000000000000000000000000000000000000000000000000000000080"
+                ]
+              },
+              {
+                "pc": 7,
+                "op": "MLOAD",
+                "gas": 79,
+                "gasCost": 3,
+                "depth": 1,
+                "stack": [
+                  "40"
+                ],
+                "memory": [
+                  "0000000000000000000000000000000000000000000000000000000000000000",
+                  "0000000000000000000000000000000000000000000000000000000000000000",
+                  "0000000000000000000000000000000000000000000000000000000000000080"
+                ]
+              },
+              {
+                "pc": 8,
+                "op": "STOP",
+                "gas": 76,
+                "gasCost": 0,
+                "depth": 1,
+                "stack": [
+                  "80"
+                ],
+                "memory": [
+                  "0000000000000000000000000000000000000000000000000000000000000000",
+                  "0000000000000000000000000000000000000000000000000000000000000000",
+                  "0000000000000000000000000000000000000000000000000000000000000080"
+                ]
+              }
+        ]
+        "#;
+    let geth_steps: Vec<GethExecStep> =
+        serde_json::from_str(input_trace).expect("Error on trace parsing");
+    let block = BlockData::new_single_tx_geth_steps(geth_steps);
+    let mut builder = CircuitInputBuilder::new(
+        block.eth_block.clone(),
+        block.block_ctants.clone(),
+    );
+    builder.handle_tx(&block.eth_tx, &block.geth_trace).unwrap();
+
+    let stack_ops = builder.block.container.sorted_stack();
+
+    let circuit = StateCircuit::<2000, 100, 2, 100, 1023, 1000> {
+        memory_ops: vec![],
+        stack_ops,
+        storage_ops: vec![],
+    };
```

Comment:
This seems like a really really small amount of data to make a "close to real-world" bench example. I would expect a loop or something that repeats the same data over and over to fill the Circuit with repeated constraints till we have an ammount that we consider close to a real-world example in mainnet.

Any ideas here @han0110 ?

Reply:
I think in our case witness assignment time would be small compared to the actual proof calculation. So I think we might just raise the `k` without dummy data to see how proving time probably goes. But it would definitely be great to use some real-world data for the real prover.